### PR TITLE
Specify shorter control_path to avoid 'unix domain socket too long' problem

### PR DESCRIPTION
**Pull request by tom-meyer**
Paired with Ryan Merriman on this fix.  This issue is reported in the following Jira:  https://issues.apache.org/jira/browse/METRON-168

### CODE REVIEW

File: metron-deployment/amazon-ec2/ansible.cfg [hunk 1]
```diff
@@ -25,4 +25,4 @@ log_path = ./ansible.log
 
 # fix for "ssh throws 'unix domain socket too long' " problem
 [ssh_connection]
-control_path = %(directory)s/%%h-%%p-%%r
+control_path = /tmp/%%h-%%p-%%r
```
Comment (dlyle65535) [inline metron-deployment/amazon-ec2/ansible.cfg]: I wouldn't use temp. DoD stigs require noexec on tmp. Alternatively, see if the socket will work when /tmp is mounted noexec.

**Comment by nickwallen** [comment]
+1 Tested successfully with EC2 deployment.  

This same issue has bit us twice now.  It was difficult to diagnose this issue because it only impacted a few users.  It seems that OpenSSH imposes a 104 character limit on the length of the path to the domain socket.  

The previous configuration embedded things like the username and working directory in the path of the domain socket.  It would fail for people who had very specific platform differences, like a long username or highly nested path to Metron, which would then exceed the 104 character limit.  I myself was never able to replicate the problem until now.

This is our second attempt at a fix and I think this is the right approach.  Using `%%C` results in a fixed-length hash. Using `~/.ssh/` also ensures that the path is also of fixed-length.  This removes any variability that might cause it to exceed 104 characters.  [Seems like others agree too.](https://github.com/ansible/ansible/issues/11536)

**Comment by cestella** [comment]
+1 this looks good

**Comment by nickwallen** [comment]
This fix was not working for @dlyle65535 today.  It would error out because it did not understand `%C`.  

I dug a little and found that this option was added in [OpenSSH 6.7](http://www.openssh.com/txt/release-6.7).  I wanted to provide a warning for other users who have older versions of OpenSSH.

**Pull request closed without merging**